velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -379,5 +379,44 @@ TEST_F(DateTimeFunctionsTest, dayofWeekTs) {
   }
 }
 
+TEST_F(DateTimeFunctionsTest, dateDiffDate) {
+  const auto dateDiff = [&](std::optional<int32_t> date1,
+                            std::optional<int32_t> date2) {
+    return evaluateOnce<int32_t, int32_t>(
+        "date_diff(c0, c1)", {date1, date2}, {DATE(), DATE()});
+  };
+
+  const auto dateDiffAlias = [&](std::optional<int32_t> date1,
+                                 std::optional<int32_t> date2) {
+    return evaluateOnce<int32_t, int32_t>(
+        "datediff(c0, c1)", {date1, date2}, {DATE(), DATE()});
+  };
+  // Check null behaviors
+  EXPECT_EQ(std::nullopt, dateDiff(1, std::nullopt));
+  EXPECT_EQ(std::nullopt, dateDiff(std::nullopt, 0));
+
+  // Simple tests
+  EXPECT_EQ(-1, dateDiff(parseDate("2019-02-28"), parseDate("2019-03-01")));
```

Comment:
@PHILO-HE Unit tests also need to be changed. The first argument is end date, and second one is start date.

Reply:
Just fixed.